undefined reference to `backtrace_symbols` (missing `-lexecinfo` on musl libc)

When upgrading the void package of xournalpp to 1.1.1 (https://github.com/void-linux/void-packages/pull/35621) we noted the following build failure using musl libc:
```
...
/usr/bin/ld: src/xoj-preview-extractor/CMakeFiles/xournalpp-thumbnailer.dir/__/util/Stacktrace.cpp.o: in function `Stacktrace::printStracktrace(std::ostream&)':
Stacktrace.cpp:(.text+0x24a): undefined reference to `backtrace'
/usr/bin/ld: Stacktrace.cpp:(.text+0x25f): undefined reference to `backtrace_symbols'
collect2: error: ld returned 1 exit status
...
```

This is caused by a missing `-lexecinfo` when linking `xournalpp-thumbnailer`. The issue is new in 1.1.1 because this binary now links with `src/util/Stacktrace.cpp`.

The following patch seems to fix the issue:
```
--- a/src/xoj-preview-extractor/CMakeLists.txt	2022-02-13 16:30:45.000000000 -0300
+++ b/src/xoj-preview-extractor/CMakeLists.txt	2022-02-14 16:03:48.561654740 -0300
@@ -35,6 +35,7 @@
   ${librsvg_LDFLAGS}
   ${Glib_LDFLAGS}
   ${ZIP_LDFLAGS}
+  ${Backtrace_LIBRARIES}
   )
 
 target_include_directories (xournalpp-thumbnailer PRIVATE ${librsvg_INCLUDE_DIRS})
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -90,7 +90,6 @@ if (WIN32)
 	find_package(ImageMagick REQUIRED COMPONENTS convert)
 else ()
 	find_package(Backtrace REQUIRED)
-	add_includes_ldflags("${Backtrace_LIBRARIES}" "${Backtrace_INCLUDE_DIRS}")
 endif ()
 
 # GLIB

--- a/src/util/CMakeLists.txt
+++ b/src/util/CMakeLists.txt
@@ -4,9 +4,10 @@ file (GLOB_RECURSE util_SOURCES *.cpp *.h)
 
 add_library (util STATIC ${util_SOURCES})
 add_dependencies(util std::filesystem)
-target_link_libraries(util PUBLIC std::filesystem cxx17)
+target_link_libraries(util PUBLIC std::filesystem cxx17 ${Backtrace_LIBRARY})
 target_compile_features(util PUBLIC ${PROJECT_CXX_FEATURES})
 target_include_directories(util PUBLIC
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+        ${Backtrace_INCLUDE_DIR}
         #$<INSTALL_INTERFACE:>  # <prefix>/include/mylib
         )

